[Safe CPP] Address warnings in PageOverlay.cpp

stwrt opened the pull request:
#### 1cca46034aa4bb22571ba0b831ad8ed59004a055
<pre>
[Safe CPP] Address warnings in PageOverlay.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=302046">https://bugs.webkit.org/show_bug.cgi?id=302046</a>
<a href="https://rdar.apple.com/problem/164124271">rdar://problem/164124271</a>

Reviewed by Chris Dumez.

Address safer cpp warnings in PageOverlay.cpp

* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations
* Source/WebCore/page/PageOverlay.cpp

Canonical link: <a href="https://commits.webkit.org/302658@main">https://commits.webkit.org/302658@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9c3017703619dae04be47f9e04d1da82e32da22a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129811 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/2072 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40668 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137200 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/81288 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/17ce9b31-578e-4e7a-8642-f4a79352437b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131682 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/2024 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1962 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98888 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/66708 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/7ca145b9-605b-4a1a-9d9a-718661fb293c) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132758 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1539 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/116260 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79569 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/be5a88b1-c3e0-4f8d-8ade-d9a41093608e) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1454 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/34392 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80474 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/109946 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34894 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139683 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1866 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1746 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107396 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1911 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112607 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107273 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27310 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1506 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/31092 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/54653 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1939 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/65308 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1753 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1788 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1862 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/PageOverlay.cpp:
```diff
@@ -94,10 +94,10 @@ IntRect PageOverlay::bounds() const
-                width -= frameView->verticalScrollbar()->width();
-            if (frameView->horizontalScrollbar())
-                height -= frameView->horizontalScrollbar()->height();
+            if (RefPtr verticalScrollbar = frameView->verticalScrollbar())
```
nit: In small scopes like this, we usually try to use smaller names, like `scrollbar` here.

cdumez commented on Source/WebCore/page/PageOverlay.cpp:
```diff
@@ -94,10 +94,10 @@ IntRect PageOverlay::bounds() const
-                height -= frameView->horizontalScrollbar()->height();
+            if (RefPtr verticalScrollbar = frameView->verticalScrollbar())
+                width -= verticalScrollbar->width();
+            if (RefPtr horizontalScrollbar = frameView->horizontalScrollbar())
```
nit: In small scopes like this, we usually try to use smaller names, like `scrollbar` here.

cdumez commented on Source/WebCore/page/PageOverlay.cpp:
```diff
@@ -186,7 +186,8 @@ void PageOverlay::drawRect(GraphicsContext& graphicsContext, const IntRect& dirt
     if (m_overlayType == PageOverlay::OverlayType::Document) {
-        if (auto* frameView = m_page->mainFrame().virtualView()) {
+        RefPtr frameView = m_page->mainFrame().virtualView();
+        if (frameView) {
```
Why not this?
```
if (RefPtr frameView = m_page->mainFrame().virtualView()) {
```

Seems better to reduce the scope as much as possible.

cdumez commented on Source/WebCore/page/PageOverlay.cpp:
```diff
@@ -200,8 +201,10 @@ bool PageOverlay::mouseEvent(const PlatformMouseEvent& mouseEvent)
-        mousePositionInOverlayCoordinates = m_page->mainFrame().virtualView()->windowToContents(mousePositionInOverlayCoordinates);
+    if (m_overlayType == PageOverlay::OverlayType::Document) {
+        RefPtr frameView = m_page->mainFrame().virtualView();
+        mousePositionInOverlayCoordinates = frameView->windowToContents(mousePositionInOverlayCoordinates);
```
Why the new local variable, why not call `protectedVirtualView()`?

cdumez reviewed (approved):
LGTM

cdumez commented on Source/WebCore/page/PageOverlay.cpp:
```diff
@@ -84,7 +84,8 @@ IntRect PageOverlay::bounds() const
         return { { }, m_overrideFrame.size() };
 
-    RefPtr frameView = m_page->mainFrame().virtualView();
+    Ref mainFrame = m_page->mainFrame();
```
Why the new local variable? Can’t we call protectedMainFrame() instead?

cdumez replied:
Same comment applies elsewhere in this PR. Unless we’re calling the protected function more than once in the same scope, we should not need a local variable.

stwrt replied:
Fixed to calling `protectedMainFrame()` across the file.

stwrt committed:
[Safe CPP] Address warnings in PageOverlay.cpp
https://bugs.webkit.org/show_bug.cgi?id=302046
rdar://problem/164124271

Reviewed by Chris Dumez.

Address safer cpp warnings in PageOverlay.cpp

* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations
* Source/WebCore/page/PageOverlay.cpp

Canonical link: https://commits.webkit.org/302658@main
